crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -209,6 +181,102 @@ impl Trade {
         })
     }
 
+    /// Protocol fee as a cut of surplus, denominated in SURPLUS token
+    ///
+    /// Protocol fee calculation logic depends if the protocol fee
+    /// is already applied to the trade or not. Since scoring module works with
+    /// trades that already have the protocol fee applied, we need to calculate
+    /// the protocol fee as an observation of already applied protocol fee.
+    ///
+    /// The protocol fee before being applied is:
+    ///    fee = surplus_before_fee * factor
+    /// The protocol fee after being applied is:
+    ///    fee = surplus_after_fee * factor'
+    /// Also:
+    ///    surplus_after_fee = surplus_before_fee - fee
+    /// So:
+    ///    factor' = fee / surplus_after_fee = fee / (surplus_before_fee - fee)
+    /// = fee / ((fee / factor) - fee) = factor / (1 - factor)
+    ///
+    /// Finally:
+    ///     fee = surplus_after_fee * factor / (1 - factor)
+
```

Comment:
Does this new line prevent this block from being associated with the function? It's really just a nit, but I do think these types of comments should go inside the method as they don't really concern the caller (it's enough for them to know that this methods returns the fee in what unit, not how it's computed).

But someone that touches the inner workings of the method should understand this

Reply:
Agreed. moved inside function